clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -144,10 +151,12 @@ public static RequestHeader parse(ByteBuffer buffer) {
             header.size = Math.max(buffer.position() - bufferStartPositionForHeader, 0);
             return header;
         } catch (UnsupportedVersionException e) {
-            throw new InvalidRequestException("Unknown API key " + apiKey, e);
+            throw new InvalidRequestException("Unknown API key " + apiKeyId, e);
         } catch (Throwable ex) {
-            throw new InvalidRequestException("Error parsing request header. Our best guess of the apiKey is: " +
-                    apiKey, ex);
+            if (ex instanceof InvalidRequestException)
```

Comment:
How about adding `InvalidRequestException` to the catch block?
```java
        } catch (UnsupportedVersionException e) {
            throw new InvalidRequestException("Unknown API key " + apiKeyId, e);
        } catch (InvalidRequestException e) {
          throw e;
        } catch (Throwable ex) {
            throw new InvalidRequestException("Error parsing request header. Our best guess of the apiKeyId is: " +
                    apiKeyId, ex);
        }
```

Reply:
Makes sense - not sure why I didn't write that in the first place!